src/main/java/ratpack/pac4j/internal/Pac4jAuthenticator.java

```diff
@@ -55,22 +53,23 @@ public Pac4jAuthenticator(String path, RatpackPac4j.ClientsProvider clientsProvi
   public void handle(Context ctx) throws Exception {
     PathBinding pathBinding = ctx.getPathBinding();
     String pastBinding = pathBinding.getPastBinding();
-
     if (pastBinding.equals(path)) {
       RatpackWebContext.from(ctx, true).flatMap(webContext -> {
         SessionData sessionData = ((RatpackSessionStore) webContext.getSessionStore()).getSessionData();
         return createClients(ctx, pathBinding).map(clients -> {
-            final String clientName = webContext.getRequestParameter(Pac4jConstants.DEFAULT_CLIENT_NAME_PARAMETER);
-            return clients.findClient(clientName);
-          }
-        ).map(
-          Types::<Client<Credentials, CommonProfile>>cast
+              final String clientName = webContext.getRequestParameter(
+                  Pac4jConstants.DEFAULT_CLIENT_NAME_PARAMETER).get();
+              final Optional<Client> client = clients.findClient(clientName);
+              if (client.isPresent()) {
+                return client.get();
+              } else {
+                throw new TechnicalException("No default client");
+              }
+            }
         ).flatMap(client ->
-          getProfile(webContext, client)
+            getProfile(webContext, client)
```

Comment:
formatting-only

Reply:
This was incorrect indentation, I think fixing it is OK?

Reply:
After looking through the rest of the project it appears it is done that way, reverting it.